py-polars/polars/functions/lit.py

```diff
@@ -114,8 +131,17 @@ def lit(
         return lit(time_int).cast(Time)
 
     elif isinstance(value, date):
-        date_int = date_to_int(value)
-        return lit(date_int).cast(Date)
+        if dtype == Datetime:
+            time_unit = getattr(dtype, "time_unit", "us") or "us"
+            dt_utc = datetime(value.year, value.month, value.day)
+            dt_int = datetime_to_int(dt_utc, time_unit)
+            expr = lit(dt_int).cast(Datetime(time_unit))
```

Comment:
Likewise here; if we init the literal with `dt_utc` we can skip the cast and have Rust handle it internally.